linux.c

```diff
@@ -0,0 +1,90 @@
+#include <string.h>
+#include <unistd.h>
+
+#include "types.h"
+
+static inline int
+thread_stats__linux (thread_stats_t *stats, size_t *len) {
+  int self = getpid();
+  clock_t clock_ticks = sysconf(_SC_CLK_TCK);
+
+  *len = 0;
```

Comment:
The initial value of `*len` is the size of `stats`, which we must ensure doesn't overflow in the loop further down. Track the number of written elements separately and only _after_ the loop assign this value to `*len`.